pointers/pointers.c

```diff
@@ -47,13 +48,12 @@ char *find_char(char *str, int c)
 */
 char *find_string(char *haystack, char *needle)
 {
-    bool matched = false;
     char *found_char;
     char *current_haystack;
 
     *found_char = find_char(haystack, *needle);
 
-    while(found_char != "\0"){
+    while(*found_char != "\0"){
         current_haystack = found_char;
         while(*needle != "\0"){
```

Comment:
What if haystack contains only a portion of needle at the end (e.g. haystack = "need", needle ="needle")?